test/functional/test_framework/key.py

```diff
@@ -356,6 +356,26 @@ def tweak_add(self, tweak):
         ret.compressed = self.compressed
         return ret
 
+    def mul(self, data):
+        assert(self.valid)
+        assert(len(data) == 32)
+        f = int.from_bytes(data, 'big')
+        assert(f < SECP256K1_ORDER and f is not None)
+        ret = ECPubKey()
+        ret.p = SECP256K1.mul([(self.p, f)])
+        ret.valid = True
+        ret.compressed = self.compressed
+        return ret
+
+    def add(self, pk):
```

Comment:
Did you consider using the [`__add__`](https://docs.python.org/3/reference/datamodel.html#object.__add__) method so users can use the more natural notation:

```
a = ECPubKey()
b = ECPubKey()
c = a + b
```

than:

```
a = ECPubKey()
b = ECPubKey()
c = a.add(b)
```

(same comment for other `add()` and `mul()` functions in this commit)

Reply:
Done. I kept add/mul for key operations for scalar bytes (e.g. tweaking a key with hash digest), but these are just wrappers for +/- to minimise code duplication.